internal/engine/configs/api.go

```diff
@@ -98,9 +100,35 @@ func toKubernetesApplyObjects(tlr tiltfile.TiltfileLoadResult) typedObjectSet {
 	return result
 }
 
+// Pulls out all the ImageMap objects generated by the Tiltfile.
+func toImageMapObjects(tlr tiltfile.TiltfileLoadResult) typedObjectSet {
+	result := typedObjectSet{}
+
+	for _, m := range tlr.Manifests {
+		for _, iTarget := range m.ImageTargets {
+			name := apis.SanitizeName(iTarget.ID().Name.String())
+			// Note that an ImageMap might be in more than one Manifest, so we
+			// can't annotate them to a particular manifest.
+			im := &v1alpha1.ImageMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+					Labels: map[string]string{
+						LabelOwnerKind: LabelOwnerKindTiltfile,
+					},
+				},
+				Spec: iTarget.ImageMapSpec,
+			}
+			result[name] = im
+		}
+	}
+	return result
+}
+
 // Fetch all the existing API objects that were generated from the Tiltfile.
 func getExistingAPIObjects(ctx context.Context, client ctrlclient.Client) (objectSet, error) {
 	result := objectSet{}
+
+	// TODO(nick): There's got to be a more generic way to do this.
```

Comment:
oy, I take it the dynamic client would be ugly to use here?

Reply:
ya, the List object interface isn't super helpful here
https://pkg.go.dev/github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource#ObjectList